shell-here.el

```diff
@@ -141,20 +141,20 @@ root."
       ;; We need to `cd'
       (when (not (string= (shell-here-stripslash
                            (expand-file-name default-directory)) target))
+        (setq target (substring target (string-match "[^\:]+\/[^\:\*\\\<\>\|]+" target)))
```

Comment:
I think it would be better to use the built-in functions for handling Tramp paths (`TRAMP-TRAMP-FILE-P` and `TRAMP-DISSECT-FILE-NAME` / or `WITH-PARSED-TRAMP-FILE-NAME`) instead of a regexp here.

Reply:
@ieure OK, updated with latest commit